bootstrap/paths_test.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+func TestCIPaths(t *testing.T) {
+	tests := []struct {
+		Name     string
+		Base     string
+		Job      string
+		Build    string
+		Expected *Paths
+	}{
+		{
+			Name:  "normal",
+			Base:  "/some/foo/base",
+			Job:   "some-foo-job",
+			Build: "1337",
+			Expected: &Paths{
+				Artifacts:   filepath.Join("/some/foo/base", "some-foo-job", "1337", "artifacts"),
+				BuildLog:    filepath.Join("/some/foo/base", "some-foo-job", "1337", "build-log.txt"),
+				Finished:    filepath.Join("/some/foo/base", "some-foo-job", "1337", "finished.json"),
+				Latest:      filepath.Join("/some/foo/base", "some-foo-job", "latest-build.txt"),
+				ResultCache: filepath.Join("/some/foo/base", "some-foo-job", "jobResultsCache.json"),
+				Started:     filepath.Join("/some/foo/base", "some-foo-job", "1337", "started.json"),
```

Comment:
The main thing I haven't decided is if I should just hardcode these strings assuming path sep is `/` or if I should update all of the test-cases below to use `filepath.Join`. I'm leaning towards the latter since it is more portable and closer matches what we expect the method to do.